rye/find-downloads.py

```diff
@@ -139,26 +146,29 @@ def _sort_key(info):
         pref = FLAVOR_PREFERENCES.index(flavor)
     except ValueError:
         pref = -1
-    return tuple, pref
+    return pref
 
 
 final_results = {}
 for py_ver, choices in results.items():
-    choices.sort(key=_sort_key)
+    choices.sort(key=_sort_key, reverse=True)
     urls = {}
     for triple, flavor, url in choices:
+        triple = tuple(triple.split('-'))
         if triple in urls:
             continue
-        urls[tuple(triple.split('-'))] = url
+        urls[triple] = url
```

Comment:
I believe the sorting and dict containing check is not correct, resulting the generated `downloads.inc` containing some outdated releases.